### PR TITLE
obs-filters: Add luma key filter

### PR DESCRIPTION
**Pull request by DDRBoxman**

### CODE REVIEW

File: plugins/obs-filters/luma-key-filter.c [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	.type                          = OBS_SOURCE_TYPE_FILTER,
+	.output_flags                  = OBS_SOURCE_VIDEO,
+	.get_name                      = luma_key_name,
+	.create                        = luma_key_create,
+	.destroy                       = luma_key_destroy,
+	.video_render                  = luma_key_render,
+	.update                        = luma_key_update,
+	.get_properties                = luma_key_properties,
+	.get_defaults                  = luma_key_defaults
+};
```
Comment (RytoEX) [inline plugins/obs-filters/luma-key-filter.c]: Style:  Nitpick, but add a new line here.

```suggestion
};

```

**Comment by Oncorporation** [comment]
I wrote a luminance filter for my plugin that seems to share a lot in common with this code. 

I added the following to increase the number of effects and simplify the testing
- inverters for color 
- inverter for alpha
- a color filter 

This effect creates an Alpha layer on live video sources and is useful when colors are not a good way to merge layers.

https://www.twitch.tv/videos/369409420 is an example of a matrix effect that I made with luminance.

I like the use of smoothstep, but it seems to have a very limited impact.

**Comment by Oncorporation** [comment]
https://clips.twitch.tv/SteamyArtsyBubbleteaCoolStoryBro  video on how to use luminance

**Comment by AlexGuo1998** [comment]
Hello. Tried your luma key filter and found an issue. Seems it would always filter out black (`#000000`) even if you set `Luma Min` to 0. Here's the picture for testing:
colortest

Here's the result:
screenshot

**Pull request merged**